safir/src/safir/metrics/_testing.py

```diff
@@ -0,0 +1,235 @@
+"""Helpers working with metrics events in unit tests."""
+
+from abc import ABC, abstractmethod
+from pprint import pformat
+from typing import Any, Generic, TypeVar
+from unittest.mock import ANY as MOCK_ANY
+
+from pydantic import BaseModel
+
+__all__ = [
+    "ANY",
+    "NOT_NONE",
+    "BaseAssertionError",
+    "NotPublishedConsecutivelyError",
+    "NotPublishedError",
+    "PublishedCountError",
+    "PublishedList",
+    "PublishedTooFewError",
+]
+
+P = TypeVar("P", bound=BaseModel)
+"""Generic event payload type."""
+
```

Comment:
Nit: Lots of extra blank lines at the start of this file, I think because of mixing variables and type variables and classes. There's no way to avoid that for `_NotNone`, but I'd lift the other variables and type variables up above the class and `NOT_NONE` definition just to compact things a bit. (We're inconsistent about whether to put globals before or after `__all__`.)